internal/jobtokens/jobtokens.go

```diff
@@ -47,14 +47,14 @@ func New(n int) (*JobTokens, error) {
 
 // Take is how a job (goroutine) can Take its turn.
 func (jt *JobTokens) Take() {
-	jt.wg.Add(1)
 	<-jt.tokens
+	jt.wg.Add(1)
```

Comment:
```suggestion
	jt.wg.Add(1)
	<-jt.tokens
```

Reply:
This actually won't matter @krisis since there is no processing without `<-chan` however what I wanted to convey here was only Add(1) only if we get to move forward with `<-chan`

Just like what we did with Done() must be called after `jt.tokens <-`